language/dotnet/build-images.md

```diff
@@ -0,0 +1,307 @@
+```
+
+Output similar to the following appears.
+
+```console
+Building...
+warn: Microsoft.AspNetCore.DataProtection.Repositories.FileSystemXmlRepository[60]
+    Storing keys in a directory '/root/.aspnet/DataProtection-Keys' that may not be persisted outside of the container. Protected data will be unavailable when the container is destroyed.
+warn: Microsoft.AspNetCore.Server.Kestrel[0]
+     Unable to bind to http://localhost:5000 on the IPv6 loopback interface: 'Cannot assign requested address'.
+info: Microsoft.Hosting.Lifetime[0]
+    Now listening on: http://localhost:5000
+info: Microsoft.Hosting.Lifetime[0]
+    Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+    Hosting environment: Development
+info: Microsoft.Hosting.Lifetime[0]
+    Content root path: /src
+```
+
+Open a web browser and access `http://localhost:5000`. The following page should appear.
+
+![image of app page](./images/dotnet-app-verify-build.png)
+
+Press Ctrl+C in the terminal window to stop the application.
+
+## Create a Dockerfile
+
+Now that our application is running properly, let’s take a look at creating a Dockerfile.
+
+{% include guides/create-dockerfile.md %}
+
+Next, we need to add a line in our Dockerfile that tells Docker what image
+we would like to use to build our application.
+
+```dockerfile
+# syntax=docker/dockerfile:1
+
+FROM mcr.microsoft.com/dotnet/sdk:6.0 as build-env
+```
+
+Docker images can be inherited from other images. Therefore, instead of creating our own image, we’ll use the official .NET SDK image that already has all the tools and packages that we need to build a .NET application.
+
+We will use a multi-stage build and define a stage for building the application.We define a `build-env` stage in our Dockerfile using `as`.
+
+> **Note**
+>
+> To learn more about multi-stage builds, see [Use multi-stage builds](../../develop/develop-images/multistage-build.md).
+
+To make things easier when running the rest of our commands, let’s create a working directory for our source files. This instructs Docker to use this path as the default location for all subsequent commands. By doing this, we do not have to type out full file paths but can use relative paths based on the working directory.
+
+```dockerfile
+WORKDIR /src
+```
+
+Although not necessary, the commands below will copy only the csproj files and then run `dotnet restore`. Each command creates a new container layer. To speed the building of containers, Docker caches these layers. Since these files won't change often, we can take advantage of the caching by copying these files and running restore as separate commands.
+
+
+```dockerfile
+COPY src/*.csproj .
+RUN dotnet restore
+```
+
+Next, you'll need to copy the rest of your source files into the image. The line below will copy the files from the `src` directory on your local machine to a directory called `src` in the image.
+
+```dockerfile
+COPY src ./src
```

Comment:
The workdir already is `/src`, so the target in the COPY line must be `.`

```suggestion
COPY src .
```

Reply:
Thank you. Will update.